src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java

```diff
@@ -0,0 +1,148 @@
+
+public class AdviewBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpAdview>> ADVIEW_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpAdview>>() {
+            };
+    private static final String ACCOUNT_ID_MACRO = "{{AccountId}}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public AdviewBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final Imp firstImp = request.getImp().get(0);
+        final ExtImpAdview extImpAdview;
+
+        try {
+            extImpAdview = mapper.mapper().convertValue(firstImp.getExt(), ADVIEW_EXT_TYPE_REFERENCE).getBidder();
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+
+        final BidRequest modifiedRequest = modifyRequest(request, extImpAdview.getMasterTagId());
+        return Result.withValue(makeHttpRequest(modifiedRequest, extImpAdview.getAccountId()));
+    }
+
+    private static BidRequest modifyRequest(BidRequest bidRequest, String masterTagId) {
+        return bidRequest.toBuilder()
+                .imp(modifyImps(bidRequest.getImp(), masterTagId))
+                .build();
+    }
+
+    private static List<Imp> modifyImps(List<Imp> imps, String masterTagId) {
+        final List<Imp> modifiedImps = new ArrayList<>(imps);
+        modifiedImps.set(0, modifyImp(imps.get(0), masterTagId));
+        return modifiedImps;
+    }
+
+    private static Imp modifyImp(Imp imp, String masterTagId) {
+        return imp.toBuilder()
+                .tagid(masterTagId)
+                .banner(resolveBanner(imp.getBanner()))
+                .build();
+    }
+
+    private static Banner resolveBanner(Banner banner) {
+        final List<Format> formats = banner != null ? banner.getFormat() : null;
+        if (CollectionUtils.isNotEmpty(formats)) {
+            final Format firstFormat = formats.get(0);
+            return firstFormat != null
+                    ? banner.toBuilder().w(firstFormat.getW()).h(firstFormat.getH()).build()
+                    : banner;
+        }
+        return banner;
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest request, String accountId) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(resolveEndpoint(accountId))
+                .headers(HttpUtil.headers())
+                .body(mapper.encode(request))
+                .payload(request)
+                .build();
+    }
+
+    private String resolveEndpoint(String accountId) {
+        return endpointUrl.replace(ACCOUNT_ID_MACRO, HttpUtil.encodeUrl(accountId));
```

Comment:
I disagree, here we are modifying url with unknown string, it can be anything and can contain any character.. i'd say it's a barrier to make sure that resulting url is ok.